### PR TITLE
Fix QML-related issues

### PR DESCRIPTION
**Pull request by Venemo**
Fixes some issues that I encountered while trying to implement some client code in QML.

### CODE REVIEW

**Comment by chriadam** [comment]
Fixes LGTM so far at least.

**Comment by Venemo** [comment]
Fixed a bunch of more stuff, and added some more QML-friendly stuff. Mostly some trivial things, but there were some more serious issues:

* Nested classes cannot be `QObject` or even `Q_GADGET`, so I had to workaround this for `Key::Identifier`. It's ugly but I didn't have a better idea.
* `QVector<>` only becomes a JavaScript `Array` implicitly with some pre-defined template arguments, but not with custom types. So I had to add an ugly hack which involves `QVariant`
* Also custom `Q_GADGET` types cannot be returned, those also had to be wrapped in `QVariant`

So, just the usual QML bulls*** that you have to deal with in every project.

On the upside, the signing flow now works with the QML API, which was previously impossible to use. I also added a new test which uses the QML-based test suite.

File: qml/Crypto/plugintypes.h [hunk 1]
```diff
@@ -51,14 +51,24 @@ class CryptoPlugin : public QQmlExtensionPlugin
...
+    Q_PROPERTY(QString defaultCryptoPluginName READ defaultCryptoPluginName)
+    Q_PROPERTY(QString defaultCryptoStoragePluginName READ defaultCryptoStoragePluginName)
 
 public:
     CryptoManager(QObject *parent = Q_NULLPTR);
-    ~CryptoManager();
+    virtual ~CryptoManager() Q_DECL_OVERRIDE;
+
+    // QML API - make static members accessible
+    QString defaultCryptoPluginName() { return DefaultCryptoPluginName; }
```
Comment (chriadam) [inline qml/Crypto/plugintypes.h]: should be const method, same for the one below

File: qml/Secrets/main.cpp [hunk 2]
```diff
@@ -59,6 +61,26 @@ Sailfish::Secrets::Plugin::SecretManager::~SecretManager()
 {
 }
 
+QString Sailfish::Secrets::Plugin::SecretManager::inAppAuthenticationPluginName() const {
```
Comment (chriadam) [inline qml/Secrets/main.cpp]: usual style is the put opening function definition brace on a new line

**Comment by chriadam** [comment]
A couple of minor style comments, aside from that LGTM.  I'll ask Bea to review also.

File: qml/Crypto/storedkeyidentifiersrequestwrapper.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+namespace Sailfish {
+
+namespace Crypto {
+
+namespace Plugin {
+
+class KeyIdentifier {
+    Q_GADGET
+    Q_PROPERTY(QString name READ name)
+    Q_PROPERTY(QString collectionName READ collectionName)
```
[the remark this answers is not in the document]
Comment (chriadam) [reply qml/Crypto/storedkeyidentifiersrequestwrapper.h]: This one is a Q_GADGET not a Q_OBJECT so they cannot notify, and the field values aren't constant.  This is a normal way to expose a value type to QML.

File: qml/Crypto/plugintypes.h [hunk 4]
```diff
@@ -51,14 +51,24 @@ class CryptoPlugin : public QQmlExtensionPlugin
 class CryptoManager : public Sailfish::Crypto::CryptoManager
 {
     Q_OBJECT
+    Q_PROPERTY(QString defaultCryptoPluginName READ defaultCryptoPluginName)
+    Q_PROPERTY(QString defaultCryptoStoragePluginName READ defaultCryptoStoragePluginName)
```
[the remark this answers is not in the document]
Comment (chriadam) [reply qml/Crypto/plugintypes.h]: These should be marked constant

File: qml/Crypto/storedkeyidentifiersrequestwrapper.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (Venemo) [reply qml/Crypto/storedkeyidentifiersrequestwrapper.h]: In this case they never change, as it doesn't have a `set` method (no point of having one), so `CONSTANT` is a good choice.

**Comment by Venemo** [comment]
@chriadam @blammit Addressed your concerns and rebased on top of current master. Please review and merge if you think it's ok now. :)

**Comment by chriadam** [comment]
LGTM, thanks.  Feel free to merge if all unit tests pass on device etc.

**Pull request merged**